app/components/setup/overview/index.js

```diff
@@ -38,21 +20,22 @@ class SetupOverview extends Component {
     };
   }
 
-  onNextPress = async () => {
-    try {
-      const params = {
-        name: this.state.name,
-        area: this.props.area,
-        userid: this.props.user.id
-      };
-      const area = await saveArea(params, this.props.user.token);
-      // TODO: save area: image in reducer
+  componentWillReceiveProps(newProps) {
+    if (newProps.areaSaved) {
```

Comment:
To avoid go to the last step and finish it directly I think we should reset the setup reducer state each time we start it.
What do you think about a reducer like this ?
``` 
if (action.type === 'INIT_SETUP') {
  return initialState;
}
```